### PR TITLE
fix(query): Disabled the shrink_scalar optimization for cast expressions during the resolve procedure args

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Explicit casts might be present to ensure the type of a procedure parameter conforms to an expected type.  Simplifying these away might lead to procedure call failures.

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/17119)

### CODE REVIEW

**Comment by sundy-li** [comment]
try fix this with modify of 
```

    fn try_fold_constant<Index: ColumnIndex>(
        &self,
        expr: &EExpr<Index>,
        enable_shrink: bool
    )
```

**Comment by BohuTANG** [comment]
Using this PR as a benchmark check (unrelated to PR changes).

File: src/query/sql/src/planner/semantic/type_check.rs [hunk 1]
```diff
@@ -5067,8 +5066,9 @@ impl<'a> TypeChecker<'a> {
     fn try_fold_constant<Index: ColumnIndex>(
         &self,
         expr: &EExpr<Index>,
+        enable_shrink: bool,
     ) -> Option<Box<(ScalarExpr, DataType)>> {
-        if expr.is_deterministic(&BUILTIN_FUNCTIONS) {
+        if expr.is_deterministic(&BUILTIN_FUNCTIONS) && enable_shrink {
```
Comment (sundy-li) [inline src/query/sql/src/planner/semantic/type_check.rs:5071]: we should always do constant folder.

enable_shrink need to be checked inside

Comment (TCeason) [reply src/query/sql/src/planner/semantic/type_check.rs:5071]: Done

**Review by sundy-li — APPROVED** [review]

**Pull request merged**